plugin/storage/cassandra/spanstore/reader.go

```diff
@@ -50,7 +50,6 @@ const (
 		SELECT trace_id
 		FROM service_name_index
 		WHERE bucket IN ` + bucketRange + ` AND service_name = ? AND start_time > ? AND start_time < ?
-		ORDER BY start_time DESC
```

Comment:
what about the other queries that do the same?

Reply:
I am actually starting to think it's not right to remove the order by. Yes, the records are already ordered on disk, but only within a partition, which includes the bucket number. Bucket numbers are used to avoid hot spots.

Reply:
As far as I understand the query is using the full partition key, I've tested this using the same query here and ensured that all the results are sorted in the correct order

Reply:
what I mean is, the partition key is a pair of service_name + bucket:

```
PRIMARY KEY ((service_name, bucket), start_time)
```

The query always runs against all buckets, so for any given service name the query checks 10 different partitions. The results are only sorted within each partition, not across the whole result set, unless we add the order-by clause.